core/BUILD.bazel

```diff
@@ -1,33 +1,23 @@
-load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
+load("//docs:stardoc.bzl", "generate")
```

Comment:
Unfortunately, this is problematic.

Separating the Stardoc parts into `//docs` means that users of `rules_nixpkgs_core` who do not depend on anything in `@rules_nixpkgs_core//docs` do not require Stardoc. In other words, moving the Stardoc dependency from `//docs` into `//` lifts Stardoc from being just a dev-dependency into being a regular dependency.

But, Stardoc is not needed for users of `rules_nixpkgs_core`, so, it should just be a dev-dependency.

Reply:
I see what you mean, but I'm not sure this is really a problem here. There is no Bazel code path that would even go from dependants into `stardoc.bzl` unless explicitly called.

Reply:
When another project references a target in this BUILD file, e.g. the `bzl_library` target, then Bazel will load this BUILD file. It will also try to load the `load` commands, which will lead it to depend on Stardoc, because `stardoc.bzl` `load`s it. `load` cannot be guarded, so, this makes it a strict dependency.

Reply:
What do you suggest? Add a `doc` package for each module and do `generate()` in there?